internal/services/network/subnet_resource.go

```diff
@@ -158,20 +159,54 @@ func resourceSubnet() *pluginsdk.Resource {
 					},
 				},
 			},
+		},
+	}
 
-			"enforce_private_link_endpoint_network_policies": {
-				Type:     pluginsdk.TypeBool,
-				Optional: true,
-				Default:  false,
-			},
+	if features.FourPointOhBeta() {
+		resource.Schema["private_endpoint_network_policies_enabled"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		}
 
-			"enforce_private_link_service_network_policies": {
-				Type:     pluginsdk.TypeBool,
-				Optional: true,
-				Default:  false,
-			},
-		},
+		resource.Schema["private_link_service_network_policies_enabled"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		}
+	} else {
```

Comment:
fwiw since 2 of these are always present, we could move the two common fields into the schema and conditionally add the older fields here?

```suggestion
	if !features.FourPointOh() {
```

Reply:
@tombuildsstuff  I did it this way so that the fields can conflict with each other and to show the change in the `default` value, which the service team requested. I would be happy to do this as you suggested, but we would need to update the `default` value for `private_endpoint_network_policies_enabled` from `true` to `false` for the 4.0 release.

Reply:
ah in that case we've been using functions on the default value for it, e.g.

```
Default: func() interface{} {
  if features.FourPointOh() {
    return "First"
  }
  return "Second"
}
```

Reply:
Ah... that makes sense, does this also work for the `ConflictsWith` schema attribute?